umbral/keys.py

```diff
@@ -29,7 +29,7 @@ def __init__(self, bn_key: CurveBN, params: UmbralParameters=None):
 
         self.params = params
         self.bn_key = bn_key
-        self.pubkey = UmbralPublicKey(self.bn_key * self.params.g)
+        self.pubkey = UmbralPublicKey(self.bn_key * self.params.g, params=params)
```

Comment:
Btw do we ever instantiate `UmbralPublicKey` directly? If not, should we make `params` a required argument?

Reply:
We do frequently instantiate `UmbralPublicKey` directly and frequently - by deserializing.

Reply:
I wouldn't call it "directly" - that's deserialization which gets called by the user. E.g. `UmbralPublicKey.from_bytes`, not `__init__` directly.
My point is that all methods which are not to be called by the user directly can make `params` a required argument, so that we don't forget to set it accidentally

Reply:
Haha, right.  It's almost like `__init__` is private.  ;-)

Reply:
Yeah, good comment. I already took care of making params a required argument in the next PR.